docs/src/pages/index.js

```diff
@@ -0,0 +1,135 @@
+import styles from './styles.module.css';
+
+function FormField({ value, type, onChange, name, id }) {
+  switch (type) {
+    case 'boolean':
+      return (
+        <>
+          <input id={id} name={name} type="checkbox" onChange={() => onChange({
+            target: {
+              name,
+              value: !value,
+            }
+          })} checked={value} />
+        </>
+      )
+    case 'number':
+      return (
+        <>
+          <input name={name} id={id} className={styles.input} type="number" onChange={onChange} value={value} />
+        </>
+      )
+    default:
+      return null;
+  }
+}
+
+function SnippetForm({ args, values, onChange }) {
+
+  const handleChange = (evt) => {    
+    onChange({
+      ...values,
+      [evt.target.name]: evt.target.value,
+    })
+  }
+
+  return (
+    <div className={styles.generatorForm}>
+      {args.map(arg => (
+        <div className={styles.annotatedLayout}>
+          <div className={styles.annotatedLayoutLabel}>
+            <h3>{arg.displayName}</h3>
+            <div>{arg.description}</div>
+          </div>
+          <div className={styles.annotatedLayoutBody}>
+            <FormField id={arg.id} name={arg.id} type={arg.type} value={values[arg.id]} onChange={handleChange} />
+          </div>
+        </div>
+      ))}
+    </div>
+  )
+}
+
+function SnippetGenerator({ template }) {
+  const [customize, showCustomizeForm] = useState(false)
+  const [snippet, updateSnippet] = useState('')
+  const [values, updateValues] = useState(template.arguments.reduce((acc, arg) => {
+    acc[arg.id] = arg.default
+
+    return acc
+  }, {}))
+
+  useEffect(() => {
+    const tpl = values.iife ? `(function () {${template.template}})();` : template.template
+
+    updateSnippet(Object.entries(values).reduce((acc, [key, val]) => {
+      return acc.replace(`{{${key}}}`, val)
+    }, tpl))
+  }, [values])
+
+  return (
+    <div className={styles.generator}>
+      <h2 className={styles.cardHeading}>{template.name}</h2>
+      <p className={styles.cardDescription}>{template.description}</p>
+      {customize && <SnippetForm args={template.arguments} values={values} onChange={updateValues} />}
+      <div className={styles.snippetWrapper}>
+        <button type="button" className={styles.customizeButton} onClick={showCustomizeForm}>Customize</button>
+        <CodeBlock className="language-js">{snippet}</CodeBlock>
+      </div>
+    </div>
+  )
+}
+
+function Home() {
+  const context = useDocusaurusContext();
+  const {siteConfig = {}} = context;
+
+  return (
+    <Layout
+      title={siteConfig.title}
+      description={siteConfig.tagline}>
+      <header className={clsx('hero hero--primary', styles.heroBanner)}>
+        <div className="container">
+          <h1 className="hero__title">{siteConfig.title}</h1>
+          <p className="hero__subtitle">{siteConfig.tagline}</p>
+        </div>
+      </header>
+      <main>
+        <section>
+          <div className="container">
```

Comment:
There is no such class 🤔  or is it from docusaurs "internal" stylesheets?

Reply:
internal docusaurus class